### ISSUE TITLE
Can not start Projects from SD Card

### ISSUE THREAD

**Issue by ementi** [nodesign/weio#258]
Like the title says, even after upgrading to V1.2, I can not start any projects from SD Card. Not even if I create a new project and try to start the new "Hello World" default main.py .

This appears after trying to start the Program:
"MODULE CAN'T BE LOADED. Maybe you have some errors in modules that you wish to import?
Traceback (most recent call last):
File "./weioRunner.py", line 258, in launcher
self.userMain = __import__(projectModule, fromlist=[''])
ImportError: No module named sd.Test_1.main "

And this after stopping:
"
WeIO user program stopped. It was running since : 2015-10-29 21:35:41
Process Process-10:
Traceback (most recent call last):
File "/usr/lib/python2.7/multiprocessing/process.py", line 258, in _bootstrap
self.run()
File "/usr/lib/python2.7/multiprocessing/process.py", line 114, in run
self._target(*self._args, **self._kwargs)
File "./weioRunner.py", line 290, in launcher
msg = self.qIn.get()
File "/usr/lib/python2.7/multiprocessing/queues.py", line 117, in get
res = self._recv()
File "./weioRunner.py", line 310, in signalHandler
if "stop" in vars(userControl.userMain):
TypeError: vars() argument must have __dict__ attribute 
"

(Can't post pictures for some reason, so I just copied the Error Message)
Sincerely ementi

**Comment by ks156** [comment]
Could you post the output of these commands :  ```ls -alR /weioUser/sd``` and ```df```

It may help.

**Comment by ks156** [comment]
Actually, there's a bug with removable devices in v1.2. Some native language are not supported.
(ref http://we-io.net/forum/viewtopic.php?f=2&t=775).

In your case, the problem was already present on v1.1, then the cause is maybe different.
Can you also tell me in which filesystem your SD card is formatted ?

**Comment by ementi** [comment]
The output:
root@WEIO:~# ls -al /weioUser/sd
drwxr-xr-x    3 root     root         32768 Oct 29 20:58 .
drwxr-xr-x    1 root     root             0 Oct 29 20:52 ..
-rwxr-xr-x    1 root     root             0 Oct 29 20:58 __init__.py
drwxr-xr-x    3 root     root         32768 Oct 29  2015 test_1

The card is formatted in FAT32

**Comment by ks156** [comment]
@ementi : can you also give me the output of : 
```ls -al /weioUser/sd/test_1``` and ```df```

That's strange, because ```init.py``` should be ```__init__.py```

Can you also try to type the following commands, and put here the output : 
```
/etc/init.d/usd stop
cd /
umount /weioUser/sd
mount /dev/sda1 /mnt
```

**Comment by ementi** [comment]
root@WEIO:~# ls -al /weioUser/sd/test_1/
drwxr-xr-x    3 root     root         32768 Oct 29  2015 .
drwxr-xr-x    3 root     root         32768 Oct 29 20:58 ..
-rwxr-xr-x    1 root     root             0 Oct 29  2015 \__init__.py
-rwxr-xr-x    1 root     root           348 Oct 29  2015 index.htm
-rwxr-xr-x    1 root     root           118 Oct 29  2015 main.py
drwxr-xr-x    6 root     root         32768 Oct 29 20:57 www

Sorry, the \__init__.py was right, It was just my second comment on github :D

root@WEIO:~# df /weioUser/sd/
Filesystem           1K-blocks      Used Available Use% Mounted on
/dev/sda1             15549952      6368  15543584   0% /weioUser/sd

Unfortunately I don't get any output by the commands:
/etc/init.d/usd stop
cd /
umount /weioUser/sd
mount /dev/sda1 /mnt

and now my SD card is empty

**Comment by ks156** [comment]
>>> and now my SD card is empty and the following command gives me this output

Simply reboot the board, or type : 
````
umount /mnt
/etc/init.dusd start
````
and then remove and plug your SD card again. Your SD card is not mounted anymore, that's all.

Everything's looks OK, so I don't understand the problem.
Could you try to create a new project on SD Card, and test again ?

**Comment by ementi** [comment]
You are my hero of the day :)
HelloWorld does work now. I have no idea how and why but it does.
Although I can't open the html files in the development environment. Which I was able to before.

**Comment by drasko** [comment]
How did you make it work?

**Comment by ks156** [comment]
I suspect some problems with this SD card (already had some inconsistant behavior with one of my card). Try to format it again, I'm almost sure it will work again like a charm.

**Comment by ementi** [comment]
Formatted my SD Card, but problem of not being able to open index.htm persists. Might it be a problem that in the file tree the name is "index.htm" instead of "index.html"?

@drasko 
I just followed the instructions given here. Nothing more. Maybe the unmounting and mounting again did the job.

**Comment by ks156** [comment]
> "index.htm" instead of "index.html"

Indeed. And this is probably related to this issue : https://github.com/nodesign/weio/issues/261

**Comment by ementi** [comment]
fix confirmed :)

**Comment by drasko** [comment]
@ks156 we close this one? #216 will be fixed soon with v1.3.

**Comment by ks156** [comment]
Yeap

**Comment by ks156** [comment]
@ementi : if you want, I can send you a new ```weio_recovery.bin``` image that should fix your problem.
Do you want to test it ? Do you know how to flash the board with a usb key ?

**Comment by ementi** [comment]
That would be great! I flashed my WEIO with V1.3 with the sd card, same procedure isn't it?

**Comment by drasko** [comment]
You mean v1.2. Yes, it will be the same procedure, described here: https://github.com/nodesign/weio/wiki/WeIO-Recovery#offline-manual-update-via-sd-card-or-usb-key

**Comment by ks156** [comment]
@ementi : Here's a link where you can download a new image that should fix the issue #216 : https://www.dropbox.com/s/cm0ip8vmqya625t/weio_recovery.bin?dl=0

Can you comment the issue #216 to let me know if it fix the problem or not ?
